amethyst_assets/src/asset.rs

```diff
@@ -163,16 +169,16 @@ pub trait Context {
     fn category(&self) -> &str;
 
     /// Provides the conversion from the data format to the actual asset.
-    fn create_asset(&self, data: Self::Data) -> Result<Self::Asset, Self::Error>;
+    fn create_asset(&self, data: Self::Data) -> Self::Result;
 
     /// Notifies about an asset load. This is can be used to cache the asset.
     /// To return a cached asset, see the `retrieve` function.
-    fn cache(&self, _spec: AssetSpec, _asset: &Self::Asset) {}
+    fn cache(&self, _spec: AssetSpec, _asset: Self::Result) {}
 
     /// Returns `Some` cached value if possible, otherwise `None`.
     ///
     /// For a basic implementation of a cache, please take a look at the `Cache` type.
-    fn retrieve(&self, _spec: &AssetSpec) -> Option<Self::Asset> {
+    fn retrieve(&self, _spec: &AssetSpec) -> Option<Self::Result> {
```

Comment:
I don't think we want the `cache` and the `retrieve` methods to deal with futures..